postgres.go

```diff
@@ -0,0 +1,374 @@
+package eventmaster
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/lib/pq"
+	_ "github.com/lib/pq"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	eventmaster "github.com/wish/eventmaster/proto"
+)
+
+// implements datastore
+type PostgresStore struct {
+	db *sql.DB
+}
+
+type PostgresConfig struct {
+	Addr        string `json:"addr"`
+	Port        int    `json:"port"`
+	Database    string `json:"database"`
+	ServiceName string `json:"service_name"`
+	Username    string `json:"username"`
+	Password    string `json:"password"`
+}
+
+func NewPostgresStore(c PostgresConfig) (*PostgresStore, error) {
+	var host string
+	if c.ServiceName != "" {
+		host = c.ServiceName + ".service.consul"
```

Comment:
In a separate PR we will want to clean up any hardcoded references to Consul

Reply:
Noted, will do in a follow-up PR